client/a8c-for-agencies/components/user-contact-support-modal-form/index.tsx

```diff
@@ -0,0 +1,232 @@
+import { Button, FormLabel } from '@automattic/components';
+import { Modal } from '@wordpress/components';
+import { Icon, close } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { ChangeEvent, FormEventHandler, useCallback, useEffect, useState } from 'react';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormSelect from 'calypso/components/forms/form-select';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import FormTextarea from 'calypso/components/forms/form-textarea';
+import { useDispatch, useSelector } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { getCurrentUser } from 'calypso/state/current-user/selectors';
+import { successNotice } from 'calypso/state/notices/actions';
+import useSubmitContactSupport from './use-submit-contact-support';
+
+import './style.scss';
+
+type Props = {
+	show: boolean;
+	onClose?: () => void;
+};
+
+const DEFAULT_MESSAGE_VALUE = '';
+const DEFAULT_PRODUCT_VALUE = 'a4a';
+
+export default function UserContactSupportModalForm( { show, onClose }: Props ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const user = useSelector( getCurrentUser );
+
+	const [ name, setName ] = useState( user?.display_name );
+	const [ email, setEmail ] = useState( user?.email );
+	const [ product, setProduct ] = useState( DEFAULT_PRODUCT_VALUE );
+	const [ site, setSite ] = useState( '' );
+	const [ message, setMessage ] = useState( DEFAULT_MESSAGE_VALUE );
+
+	const isPressableSelected = product === 'pressable';
+	const hasCompletedForm = !! message && !! name && !! email && !! product && ! isPressableSelected;
+
+	const { isSubmitting, submit, isSubmissionSuccessful } = useSubmitContactSupport();
+
+	const onModalClose = useCallback( () => {
+		setMessage( DEFAULT_MESSAGE_VALUE );
+		setProduct( DEFAULT_PRODUCT_VALUE );
+		onClose?.();
+
+		dispatch( recordTracksEvent( 'calypso_a4a_user_contact_support_form_close' ) );
+	}, [ dispatch, onClose ] );
+
+	useEffect( () => {
+		if ( isSubmissionSuccessful ) {
+			dispatch(
+				successNotice( translate( 'Thank you! Support request submitted.' ), {
+					id: 'submit-product-contact-support-success',
+					duration: 5000,
+				} )
+			);
+			onModalClose();
+		}
+	}, [ dispatch, isSubmissionSuccessful, onModalClose, translate ] );
+
+	const onNameChange = useCallback( ( event: ChangeEvent< HTMLInputElement > ) => {
+		setName( event.currentTarget.value );
+	}, [] );
+
+	const onEmailChange = useCallback( ( event: ChangeEvent< HTMLInputElement > ) => {
+		setEmail( event.currentTarget.value );
+	}, [] );
+
+	const onSiteChange = useCallback( ( event: ChangeEvent< HTMLInputElement > ) => {
+		setSite( event.currentTarget.value );
+	}, [] );
+
+	const onProductChange: FormEventHandler = ( { target } ) => {
```

Comment:
Nitpicking: For consistency, we can use useCallback here since we use it on the rest of the handlers.

Reply:
Yes, good point. Thank you!